client/my-sites/plan-price/index.jsx

```diff
@@ -19,11 +19,33 @@ export class PlanPrice extends Component {
 			displayPerMonthNotation,
 			displayShortPerMonthNotation,
 			originalPricePrefix,
+			productDisplayPrice,
 			isOnSale,
 			taxText,
 			translate,
 		} = this.props;
 
+		const classes = classNames( 'plan-price', className, {
+			'is-original': original,
+			'is-discounted': discounted,
+		} );
+
+		// productDisplayPrice is returned from Store Price and provides a geo-IDed currency format
+		if (
+			( productDisplayPrice && ! Array.isArray( rawPrice ) ) ||
```

Comment:
I'm not entirely sure how to write this conditional. May need additional testing to ensure if productDisplayPrice is not present, or, perhaps some other mixture of rawPrice (with say a string, object, array), that these conditions work as expected.

Reply:
I think that the first condition (if `productDisplayPrice ` is set and `rawPrice` is not set) makes total sense, but I'm not sure of the purpose of the second condition. If I'm reading it right, it says that `<PlanPrice rawPrice=[ [ 5 ] ] />` will try to display `productDisplayPrice`, which does not exist.

What if we just said, if `productDisplayPrice` is set, we always use that?

```js
if ( productDisplayPrice ) {
```

Reply:
...Although I guess if you want your final test to pass (`will use rawPrice when rawPrice is passed an array with length > 1`) you'd need something like

```js
if ( productDisplayPrice ) {
  if ( ! rawPrice || ( Array.isArray( rawPrice )  ) {
```
```

Reply:
Not necessary, but when conditions like this get too complex, it can often be helpful to extract them into a function of their own to make them more readable, like maybe:

```js
const shouldUseDisplayPrice = () => {
  if ( ! productDisplayPrice ) {
    return false;
  }
  if ( rawPrice ) {
    if ( Array.isArray( rawPrice ) && rawPrice.length > 1 ) {
      return false;
    }
  }
  return true;
};

if ( shouldUseDisplayPrice() ) {
```

Reply:
Added this in, I like the idea of moving this to a function to remove the line of conditionals, much easier to understand and read. I also tested a couple of use cases manually and also ensured the tests worked too.